.github/workflows/rust-integration-tests.yml

```diff
@@ -0,0 +1,33 @@
+name: rust-integration-tests
+on:
+  schedule:
+    - cron: "0 8,20 * * *"
+  push:
+  workflow_dispatch:
+
+jobs:
+  integration:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+
+      - name: build beerus
+        run: cargo build -p beerus-cli --release
+
+      - name: start beerus RPC on MAINNET
+        env:
+          BEERUS_VERSION: "ci"
+          NETWORK: "mainnet"
+          ETH_EXECUTION_RPC: ${{ secrets.ETH_EXECUTION_RPC_MAINNET }}
+          STARKNET_RPC: ${{ secrets.STARKNET_RPC_0_6_0_MAINNET }}
+        run: ./target/release/beerus &
```

Comment:
the readiness probe might be a solution for the race condition (e.g. when tests start when Beerus is not yet up); or a fixture that ensures that Beerus is up before each test (not necessary in the same PR though)

the quick & dirty way from integration-test.sh: https://github.com/eigerco/beerus/blob/main/scripts/integration-test.sh#L12

Reply:
Yes that's was I was looking into, but for some reason I didn't think of using this solution. It should work!

Reply:
I second that! Readiness probes are great solution for that. I always used [this script](https://github.com/vishnubob/wait-for-it) for that kind of checks.

Reply:
Simple `GET /ready` handler should do here, as the server is started after Helios is synced up to a checkpoint.

Reply:
I'm currently testing this [on my dedicated fork.](https://github.com/pierre-l/beerus-ci-test/actions). (because scheduled runs only work on the default branch)
I'll set this thread as resolved if it passes >95% of the time overnight.